lib/sshutils/scp/scp.go

```diff
@@ -386,23 +389,25 @@ func (cmd *command) serveSink(ch io.ReadWriter) error {
 	// directory.
 	if cmd.Flags.DirectoryMode {
 		if len(cmd.Flags.Target) != 1 {
-			return trace.BadParameter("in directory mode, only single upload target is allowed but %v provided", len(cmd.Flags.Target))
+			return trace.BadParameter("in directory mode, only single upload target is allowed but %q provided",
+				cmd.Flags.Target)
 		}
-
-		fi, err := os.Stat(cmd.Flags.Target[0])
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		if mode := fi.Mode(); !mode.IsDir() {
+		if !cmd.FileSystem.IsDir(cmd.Flags.Target[0]) {
 			return trace.BadParameter("target path must be a directory")
 		}
 	}
 
+	rootDir := localDir
+	if len(cmd.Flags.Target) != 0 && cmd.FileSystem.IsDir(cmd.Flags.Target[0]) {
```

Comment:
Could you add a comment here about what this condition checks and why.

Reply:
I'll move that to a self-explanatory helper